### PR TITLE
Deprecate a number of inplace methods

### PR DESCRIPTION
**Pull request by jabooth**
We've steadily been moving away from mutating our objects in place, prefering to, where possible, return new objects with the desired state:
```py
a = ...
a.foo()
# a is changed!
```
```py
a = ...
b = a.foo()
# a is unchanged!
```
We try and mark mutating operations with having `inplace` in the method name. This PR deprecates a number of these:

- `apply_inplace()*`
- `from_vector_inplace()*` 
- `normalize_std_inplace()`
- `normalize_norm_inplace()`

`*` in these cases the inplace variant still exists, but is now a private (`_`-prepended) method. We still use these methods in performance critical sections of menpo, but we are now strongly discouraging others from using them. 

This PR should be brought in after #651.

### CODE REVIEW

**Comment by jabooth** [comment]
@menpobot test this please

File: menpo/base.py [hunk 1]
```diff
@@ -90,6 +87,23 @@ def _as_vector(self, **kwargs):
 
     def from_vector_inplace(self, vector):
         """
+        Deprecated. Use hte non-mutating API, :map:`from_vector`.
```
Comment (patricksnape) [inline menpo/base.py]: hte

File: menpo/base.py [hunk 2]
```diff
@@ -90,6 +87,23 @@ def _as_vector(self, **kwargs):
...
+
+        For internal usage in performance-sensitive spots,
+        use, see `_from_vector_inplace()`
```
Comment (patricksnape) [inline menpo/base.py]: ,use, see

**Comment by patricksnape** [comment]
+1 looks good, just small typos.

**Comment by patricksnape** [comment]
Happy once Appveyor is done +1

**Pull request merged**